src/haddock/modules/analysis/ilrmsdmatrix/ilrmsd.py

```diff
@@ -0,0 +1,324 @@
+            r_start, r_end = chain_ranges[r_chain]
+            l_starts = [chain_ranges[l_chain][0] for l_chain in l_chains]
+            l_ends = [chain_ranges[l_chain][1] for l_chain in l_chains]
+            # put system at origin of the receptor interface
+            Q_r_int = Q_int[r_start: r_end + 1]
+            P_r_int = P_int[r_start: r_end + 1]
+            Q_int = Q_int - centroid(Q_r_int)
+            P_int = P_int - centroid(P_r_int)
+            # put interfaces at the origin
+            # find the rotation that minimizes the receptor interface rmsd
+            Q_r_int = Q_int[r_start: r_end + 1]
+            P_r_int = P_int[r_start: r_end + 1]
+            U_int = kabsch(P_r_int, Q_r_int)
+            P_int = np.dot(P_int, U_int)
+            
+            # Identify ligand coords concatenating all the ligand chains
+            Q_l_int = np.empty((0, 3))
+            P_l_int = np.empty((0, 3))
+            for l_st, l_end in zip(l_starts, l_ends):
+                Q_l_int = np.concatenate((Q_l_int, Q_int[l_st: l_end + 1]))
+                P_l_int = np.concatenate((P_l_int, P_int[l_st: l_end + 1]))
+            # # this will be the interface-ligand-rmsd
+            ilrmsd = calc_rmsd(P_l_int, Q_l_int)
+            return ilrmsd
+
+    def run(self):
+        """Run calculations."""
+        # initialising the number of pairs
+        ref = self.start_ref
+        mod = self.start_mod
+        nmodels = len(self.model_list)
+        for n in range(self.npairs):
+            # calculating RMSD
+            ilrmsd = self.calc_ilrmsd(ref, mod)
+            if ilrmsd is None:
+                break
+            # saving output (adding one for consistency with clusterfcc)
+            self.data[n, 0] = ref + 1
+            self.data[n, 1] = mod + 1
+            self.data[n, 2] = ilrmsd
+            # updating indices
+            if mod == (nmodels - 1):
+                ref += 1
+                mod = ref + 1
+            else:
+                mod += 1
+
+    def output(
+            self,
+            ):
+        """Write down the RMSD matrix."""
+        output_fname = Path(self.path, self.output_name)
+        # check if there are very low values in the RMSD vector
+        check_low_values = np.isclose(
+            self.data[:, 2],
+            np.zeros(self.npairs),
+            atol=0.1
+            ).any()
+        if check_low_values:
+            log.warning(f"core {self.core}: low values of RMSD detected.")
+        with open(output_fname, "w") as out_fh:
+            for data in list(self.data):
+                data_str = f"{data[0]:.0f} {data[1]:.0f} {data[2]:.3f}"
+                data_str += os.linesep
+                out_fh.write(data_str)
+
+
+def get_pair(nmodels, idx):
+    """Get the pair of structures given the 1D matrix index."""
+    if (nmodels < 0 or idx < 0):
+        err = "get_pair cannot accept negative numbers"
+        err += f"Input is {nmodels} , {idx}"
+        raise ValueError(err)
```

Comment:
Strange to have this validation here, what would trigger a negative input?

Reply:
it should be impossible, but since a negative number could mess up things badly I thought to put an extra validation step there